packages/modeling/src/operations/extrusions/slice/repairSlice.js

```diff
@@ -0,0 +1,47 @@
+const vec3 = require('../../../maths/vec3')
+
+/**
+ * Mend gaps in a 2D slice to make it a closed polygon
+ */
+const repairSlice = (slice) => {
+  if (!slice.edges) return slice
+  const vertexMap = {} // string key to vertex map
+  const edgeCount = {} // count of (in - out) edges
+  slice.edges.forEach((edge) => {
+    const inKey = edge[0].toString()
+    const outKey = edge[1].toString()
+    vertexMap[inKey] = edge[0]
+    vertexMap[outKey] = edge[1]
+    edgeCount[inKey] = (edgeCount[inKey] || 0) + 1 // in
+    edgeCount[outKey] = (edgeCount[outKey] || 0) - 1 // out
+  })
+  // find vertices which are missing in or out edges
+  const missingIn = Object.keys(edgeCount).filter((e) => edgeCount[e] < 0)
+  const missingOut = Object.keys(edgeCount).filter((e) => edgeCount[e] > 0)
+  // pairwise distance of bad vertices
+  missingIn.forEach((key1) => {
+    const v1 = vertexMap[key1]
+    // find the closest vertex that is missing an out edge
+    let bestDistance = Infinity
+    let bestReplacement
+    missingOut.forEach((key2) => {
+      const v2 = vertexMap[key2]
+      const distance = Math.hypot(v1[0] - v2[0], v1[1] - v2[1])
+      if (distance < bestDistance) {
+        bestDistance = distance
+        bestReplacement = v2
+      }
+    })
+    console.log(`Repair gap ${v1} → ${bestReplacement} dist ${bestDistance}`)
```

Comment:
console

Reply:
I thought about this one a fair amount. I agree that I don't like having console output mixed into a library. However, my reasoning was that I felt a user might want to know that modification of their design had happened. Especially with "repair" being optional now (based on hrg's feedback), a user can use the presence of the console.log to know whether it is safe to disable repair checks. Open to suggestions here.

Reply:
maybe have a log function as part of jscad internall that can be called, so such king logging could be caught and used in different ways. The code inside taht utility would disable linter(for the console log inside) because it is expected. 

We would then have ability to to log stuff but also have linter warn of unintended lefotever console calls that happen during testing stuff while developing.

Reply:
Ok. Then let’s give a little more context on what’s being repaired.

Should this use console.warning()?

@hrgdavor has a nice idea, but that functionality should be added via a separate PR, which might generate  a little discussion as well. ;)